### PR TITLE
Add information about unicode_literals and byte strings.

### PR DESCRIPTION
**Pull request by brunogola**
Scope
=====

Adding information regarding the use of `from __future__ import unicode_literals` and about explicitly marking all byte strings using the `b'byte string'` syntax.

How to try and test the changes
===============================

reviewers: @adiroiban @alibotean 

Please check if it makes sense.

### CODE REVIEW

File: content/python.rst [hunk 1]
```diff
@@ -106,6 +106,12 @@ NO
     other_bad = "string"
     some_bad = "string 'b' yes"
 
+* Use `from __future__ import unicode_literals` for Python3 compatibility.
```
Comment (adiroiban) [inline content/python.rst]: use where? how?

I think that we begin by using it in all tests... so also update the documentation so that all tests are done with unicode literals and absolute imports

the python-futures.org doc talks about cons and pros... but I think that we should also have at least a phrase arguing why we went for this

**Comment by brunogola** [comment]
> I think that we begin by using it in all tests... so also update the documentation so that all tests are done with unicode literals and absolute imports

I agree with that but I'm not sure if the styleguide is where we should mention where we begin doing it. I think that the styleguide should document how our code is meant to be written, and for now we want to use `unicode_literals` everywhere... but the strategy for updating the code could be documented somewhere else maybe

thanks

**Comment by brunogola** [comment]
needs-review

**Comment by adiroiban** [comment]
thanks!

agree that we don't need to go to deep about our transitional testing strategy

but for the testing code we can start using this right away

 changes-approved

**Comment by adiroiban** [comment]
did @alibotean reviewed these changes?

**Comment by brunogola** [comment]
ah, not really. I forgot he was included in the reviewers. sorry.

@alibotean if you have any suggestions please let me know.

**Comment by alibotean** [comment]
I'm getting a site not available for: https://python-future.org/compatible_idioms.html#strings-and-bytes

Is it just me?

File: content/python.rst [hunk 2]
```diff
@@ -106,6 +106,14 @@ NO
     other_bad = "string"
     some_bad = "string 'b' yes"
 
+* As we look forward to support Python 3, we should use
+  `from __future__ import unicode_literals`, as it helps with readability,
+  instead of explicitly having to mark every text string as `u'unicode'`.
+  More details here:
+  https://python-future.org/compatible_idioms.html#strings-and-bytes
+
+* All byte strings should be explicitly marked as in `b'byte string`.
```
Comment (alibotean) [inline content/python.rst]: This is incorrect: it should be `b'byte string'`. It's missing the final single quote.
